tests/integration/security/sds_tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,286 @@
+func TestMutualTlsOrigination(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.egress.mtls.sds").
+		Run(func(ctx framework.TestContext) {
+
+			var (
+				credNameGeneric    = "mtls-credential-generic"
+				credNameNotGeneric = "mtls-credential-not-generic"
+				fakeCredNameA      = "fake-mtls-credential-a"
+				fakeCredNameB      = "fake-mtls-credential-b"
+				credNameMissing    = "mtls-credential-not-created"
+				simpleCredName     = "tls-credential-simple-cacert"
+			)
+
+			var credentialASimple = sdstlsutil.TLSCredential{
+				CaCert: sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+
+			var credentialAGeneric = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.MustReadCert(t, "cert-chain.pem"),
+				PrivateKey: sdstlsutil.MustReadCert(t, "key.pem"),
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+
+			var credentialANonGeneric = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.MustReadCert(t, "cert-chain.pem"),
+				PrivateKey: sdstlsutil.MustReadCert(t, "key.pem"),
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Configured with an invalid ClientCert
+			var credentialBCert = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.FakeCert,
+				PrivateKey: sdstlsutil.MustReadCert(t, "key.pem"),
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Configured with an invalid ClientCert and PrivateKey
+			var credentialBCertAndKey = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.FakeCert,
+				PrivateKey: sdstlsutil.FakeKey,
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Add kubernetes secret to provision key/cert for gateway.
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{credNameGeneric}, "MUTUAL", credentialAGeneric, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{credNameGeneric})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{credNameNotGeneric}, "MUTUAL", credentialANonGeneric, true)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{credNameNotGeneric})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{fakeCredNameA}, "MUTUAL", credentialBCert, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{fakeCredNameA})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{fakeCredNameB}, "MUTUAL", credentialBCertAndKey, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{fakeCredNameB})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{simpleCredName}, "SIMPLE", credentialASimple, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{simpleCredName})
+
+			internalClient, externalServer, _, serverNamespace := sdstlsutil.SetupEcho(t, ctx)
+
+			// Set up Host Namespace
+			host := "server." + serverNamespace.Name() + ".svc.cluster.local"
+
+			testCases := map[string]struct {
+				response        []string
+				credentialToUse string
+				gateway         bool // true if the request is expected to be routed through gateway
```

Comment:
We only want to test request routed through gateway. Not clear why do we want to set this to false.

Reply:
Expectation is for all requests to go through Gateway but in case request is not routed succesfully i.e. we don't get proper response back from echoboot, it won't contain the header we send as a request from the gateway. For failing tests we have to disable this check, so it's only to make sure Gateway is actually originating the requests but only in case of successful requests